pennylane/_queuing.py

```diff
@@ -118,8 +118,12 @@ def remove(cls, obj):
 
     @classmethod
     def update_info(cls, obj, **kwargs):
-        """Updates information of an object in the queue."""
-        cls.active_context().update_info(obj, **kwargs)
+        """Updates information of an object in the active queue."""
+        cls.active_context()._update_info(obj, **kwargs)  # pylint: disable=protected-access
+
+    def _update_info(self, obj, **kwargs):
+        """Updates information of an object in the queue instance."""
+        raise NotImplementedError
```

Comment:
Changed this so that queues would have to explicitly implement `_update_info`.

This comes with: 
* `qml.QueuingContext.update_info` will work as expected
* `qml._queuing.AnnotatedQueue().update_info` will error (`cls.active_context()` returns `None`) and hence  `qml._queuing.AnnotatedQueue()._update_info` has to be called